game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,106 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  enum DiceType {
```

Comment:
Second point, enums can sometimes have issues when types are added in the future or removed (compatibility issues). In specific one can have issues with the 'valueOf' for types that are unknown. What are your thoughts to the future of this list and implications to compatibility issues that could come up?

Reply:
I like enums, but we have been bitten before. An 'unknown' or 'other' enum type can work as well. I don't have enough context to know which would work better. Typically enum's are good when you know the full set and it will never change (so perhaps argument for a string type, or maybe better yet a value object that wraps a string)

Reply:
I've changed `DiceType` to a string value object - https://github.com/triplea-game/triplea/pull/8646/commits/be242593d3e6af671eb33c661fb8590c34264ecd